spec/lib/que/worker_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Que::Worker do
+  describe ".work" do
+    subject { described_class.new.work }
+
+    it "works a job if there's one to work" do
+      FakeJob.enqueue(1)
+
+      expect(subject).to eq(:job_worked)
+      expect(FakeJob.log).to eq([1])
+      expect(QueJob.count).to eq(0)
+    end
+
+    it "returns job_not_found if there's no job to work" do
+      expect(subject).to eq(:job_not_found)
+    end
+
+    context "if the job raises an exception" do
+      it "rescues it" do
+        ExceptionalJob.enqueue(1)
+
+        expect(subject).to eq(:job_worked)
+      end
+
+      context "if the job has a failure handler" do
```

Comment:
We tend to write contexts as "**when** the cat boops the blip" (as opposed to "**if** the boop do the do"). Worth bringing that pattern here for consistency? I'm aware that the 'legacy' Que specs are "if"-style so not too fussed!